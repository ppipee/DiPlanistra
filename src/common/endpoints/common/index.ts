const commonEndpoints = {
	cities: () => '/cities',
	categories: () => '/categories',
}

export default commonEndpoints
