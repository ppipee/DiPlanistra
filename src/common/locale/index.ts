export const HOUR = [
	'{hour} ชั่วโมง',
	`{hour} {hour,plural,
    =0 {hour}
    =1 {hour}
    other {hours}}`,
]

export const DAY = [
	'{day} วัน',
	`{day} {day,plural,
    =0 {day}
    =1 {day}
    other {days}}`,
]

export const LOCALE_ALL = ['ทั้งหมด', 'All']

export const KM = ['{distance} กม.', '{distance} km.']

export const LOCALE_SAVE = ['บันทึก', 'Save']
export const LOCALE_CANCEL = ['ยกเลิก', 'Cancel']
