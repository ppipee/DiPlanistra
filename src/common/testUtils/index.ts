export { default as mockComponent } from './mockComponent'
