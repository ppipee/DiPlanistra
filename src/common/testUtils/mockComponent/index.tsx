import React from 'react'

function mockComponent(
	componentName: string,
): [({ className, children, ...props }: Record<string, any>) => any, jest.Mock<any, any>] {
	const propsSpy = jest.fn()

	const MockComponent = ({ className, children, ...props }: Record<string, any>) => {
		propsSpy(props)
		const child = children || componentName

		return (
			<div onClick={props.onClick} data-testid={props['data-testid']}>
				{child}
			</div>
		)
	}

	return [MockComponent, propsSpy]
}

export default mockComponent
