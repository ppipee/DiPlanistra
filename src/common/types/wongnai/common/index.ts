import { BasicPhoto } from '../image'

export interface LocalizeName {
	primary: string
	thai?: string
	english?: string
}

export interface Region {
	id: number
	url: string
	name: string
	welcomeUrl: string
	shortName?: string
	nameOnly: LocalizeName
	type: NameValue<number>
	country?: string
	city?: string
	cityName?: string
	cityUrl?: string
	cityWelcomeUrl?: string
	district?: string
	districtName?: string
	subDistrict?: string
	subDistrictName?: string
	coordinate: LatLng
	numberOfBusinesses: number
	numberOfRestaurants: number
	numberOfBeautyBusinesses: number
	numberOfHotels: number
	numberOfAttractions: number
	knownLocation: number
	featured: boolean
	featuredTarget?: string
	featuredMessage?: string
	iconFeatured?: string
	subRegions?: Region[]
	aliases?: string[]
	coverPicture: BasicPhoto
}

export interface LatLng {
	lat: number
	lng: number
}

export interface RegionName {
	regionName: string
	name: string
	path: string
	url: string
	params: (NameValue<string> | NameValueDescription)[]
	entities: RegionName[]
}

interface NameValueDescription {
	name: string
	value: string
	description: string
}

export interface Time {
	iso: string
	full: string
	timePassed: string
}

export interface NameValue<T> {
	name: string
	value: T
}
