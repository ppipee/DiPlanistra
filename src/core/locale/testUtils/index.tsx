import React, { ReactNode } from 'react'

import LocaleContext from '../context/LocaleContext'
import { Locale } from '../types'

interface WrapperProps {
	children: ReactNode
}

// Any because wrapper type @testing-library/react has invalid wrapper type
export function createI18nWrapper(locale: Locale): any {
	const setLocaleContextSpy = jest.fn()
	return ({ children }: WrapperProps) => (
		<LocaleContext.Provider value={{ locale, setLocaleContext: setLocaleContextSpy }}>
			{children}
		</LocaleContext.Provider>
	)
}
