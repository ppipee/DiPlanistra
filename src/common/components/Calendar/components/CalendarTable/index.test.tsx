import React from 'react'

import { render } from '@testing-library/react'
import 'jest-styled-components'

import { mockComponent } from 'common/testUtils'

import * as CalendarTableType from '.'

describe('Calendar > CalendarTable', () => {
	const useCalendarContextSpy = jest.fn()
	const CalendarRowSpy = jest.fn()
	const generateCalendarTableSpy = jest.fn()
	const [HeaderCalendarRowSpy, HeaderCalendarRowPropsSpy] = mockComponent('HeaderCalendarRow')

	jest.doMock('common/components/Calendar/HeaderCalendarRow', () => HeaderCalendarRowSpy)
	jest.doMock('common/components/Calendar/utils/generateCalendarTable', () => generateCalendarTableSpy)
	jest.doMock('common/components/Calendar/hooks/useCalendarContext', () => useCalendarContextSpy)
	jest.doMock('common/components/Calendar/components/CalendarRow', () => (props: Record<string, any>) => {
		CalendarRowSpy(props)

		return <div {...props}>{props.children}</div>
	})

	const { default: CalendarTable } = require('.') as typeof CalendarTableType

	it('should perform function CalendarTable correctly', () => {
		generateCalendarTableSpy.mockReturnValue([[1, 2, 3, 4, 5, 6, 7]])
		useCalendarContextSpy.mockReturnValue({ displayState: 'displayState' })

		render(<CalendarTable />)

		expect(CalendarRowSpy).toBeCalledTimes(1)
		expect(HeaderCalendarRowPropsSpy).toBeCalledTimes(1)
		expect(useCalendarContextSpy).toBeCalledTimes(1)

		expect(generateCalendarTableSpy).toBeCalledTimes(1)
		expect(generateCalendarTableSpy).toBeCalledWith('displayState')
	})

	it('should render CalendarTable correctly', () => {
		generateCalendarTableSpy.mockReturnValue([[1, 2, 3, 4, 5, 6, 7]])
		const { container } = render(<CalendarTable />)

		expect(container.firstChild).toMatchSnapshot()
	})
})
