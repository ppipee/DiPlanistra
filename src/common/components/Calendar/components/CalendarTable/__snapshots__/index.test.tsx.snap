// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar > CalendarTable should render CalendarTable correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  padding: 0;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > *:not(:last-child) {
  margin-bottom: 2px;
}

<div
  class="c0"
>
  <div>
    HeaderCalendarRow
  </div>
  <div
    row="1,2,3,4,5,6,7"
  />
</div>
`;
